File: Grading Students/grading-students.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 13.07.18
 * Time: 11:13
 */

/**
 * Grading students function
 *
 * @param array $grades
 * @return array
 */
function gradingStudents(array $grades): array
{

    return array_map(function ($grade) {
        $multiple = $grade + (5 - $grade % 5);
        return $grade >= 38 && $multiple - $grade < 3 ? $multiple : $grade;
    }, $grades);
}

echo "<pre>gradingStudents([73, 67, 38, 33])\r\n";
var_dump(gradingStudents([73, 67, 38, 33])); // [75, 67, 40, 33]
echo "</pre>";

File: Grading Students/grading-students.py

def gradingStudents(grades):
    return map(check_multiple, grades)


def check_multiple(grade):
    multiple = grade + (5 - grade % 5)
    return multiple if grade >= 38 and multiple - grade < 3 else grade


print(" ".join(map(str, gradingStudents([73, 67, 38, 33]))))